Feature/app scaffold

kapsterdam opened the pull request:
### ✨ Highlights – Backend & API

* **Unified `/chat` handler**

  * Accepts **JSON *or* multipart** requests.
  * Added local Whisper transcription when an `audio/*` part is present.
  * Persists every turn to **PostgreSQL** (`chat_log`).

* **Similarity search upgrade (`/analyze`)**

  * New **`top_k` query‑param** – returns up to *N* best–matching intents, capped automatically.
  * Robust MIME check & clearer error paths.

* **“Yap” conversational flow (scenario 2)**

  * Endpoints: `/yap`, `/yap/start`, `/yap/next`.
  * Dual‑agent negotiation (burger ⇄ gemeente) with draft generation.
  * Sessions live in RAM but every utterance is mirrored to **`yap_log`** in Postgres.

* **LangChain refactor** (`templates.py`)

  * `make_chain()` now receives the session dict to keep the running **draft** in sync.
  * Strict JSON enforcement via `JsonOutputParser` + `PydanticOutputParser`.

* **Dynamic intent modelling** (`models.py`, `intents.py`)

  * Checklist booleans, `draft`, and `vragen` list generated on‑the‑fly per intent.

* **Infrastructure helpers**

  * **Dockerfile** for a local PostgreSQL 16 instance.
  * Async connection pool (`asyncpg`) with auto‑migration on startup.

* **Documentation**

  * README sections for new parameters, audio upload, and Yap endpoints.

---

* ** To do **

* Modify prompts and re-test
* Change the whisper model AKA put .env variables in the init.py
* Language parameter support

ShaylaJan committed:
docs: Add background to readme

kapsterdam committed:
API Scaffolding for endpoints

kapsterdam committed:
second endpoint

kapsterdam committed:
fixed small issue with to_message

kapsterdam committed:
Fixed broken import

kapsterdam committed:
Updated response format with draft

kapsterdam committed:
updated README

kapsterdam committed:
Corrected some minor mistakes

kapsterdam committed:
Updated /chat with audio capabilities.

kapsterdam committed:
Added database connector

kapsterdam committed:
added dockerfile

jurb committed:
ignore node_modules and .next

jurb committed:
first commit frontend

jurb committed:
update readme

kapsterdam committed:
Added support for multiple intents

jurb committed:
Merge branch 'feature/app-frontend' into feature/app-scaffold

jurb committed:
add audio chat endpoint wrapper

jurb committed:
do not ignore svelte /lib 😅

jurb committed:
accept different endpoints for the recording button

jurb committed:
add an endpoint test page

jurb committed:
one .gitignore per project

jurb committed:
WIP construct layout

jurb committed:
add api store and apidebugger component

jurb committed:
add figma screenshots

jurb committed:
show more matches

jurb committed:
fix card css

jurb committed:
add localstorage

jurb committed:
add localstorage clear button

jurb committed:
blow up svg border

jurb committed:
listen for a similarity and redirect if it's below 0.5

jurb committed:
chevron left was feeling lonely

jurb committed:
Update communication style guidelines in CLAUDE.md

- Add communication style section to avoid overly enthusiastic responses
- Reduce excessive use of "Perfect!" and similar phrases
- Focus on being helpful and direct rather than overly positive

jurb committed:
Add ButtonSketchySecondary component and update choose page

- Create ButtonSketchySecondary component with white background and icon support
- Add back-arrow.svg icon for navigation
- Replace hardcoded back button with reusable component
- Update choose page to use new component with proper alignment
- Add clearApiResponses() call before navigation to reset state

jurb committed:
Add session management with localStorage persistence

- Create sessionStore.ts for managing session_id across the app
- Add localStorage persistence for session data
- Include setSessionId, getSessionId, and clearSession functions
- Handle SSR safety with browser environment checks
- Store session_id with timestamp for tracking

jurb committed:
Add intentcode and session_id support to API endpoints

- Update chat endpoint to accept and forward intentcode and session_id
- Update chatAudio endpoint to handle intentcode in FormData
- Modify SingleRecordingSection to include intentcode and session_id
- Add dynamic field naming for different endpoints (file vs audio)
- Include session management integration for chatAudio requests

jurb committed:
Convert construct route to dynamic slug-based routing

- Move construct page to /construct/[slug] structure
- Extract intentcode from URL slug parameter
- Update navigation to use intentcode as slug
- Add dynamic intent matching from analyze endpoint data
- Update page heading to use intent.intent from matched data
- Integrate session_id capture and usage in chat calls
- Fix deprecated page import to use /state

jurb committed:
Add button-sketchy-secondary-small.svg icon

- Add secondary button icon for consistent styling

jurb committed:
Update UI components and add draft text display
- Format ApiDebugger.svelte with prettier
- Disable checkbox interactions in ChecklistCard
- Update Pill component styling with proper font
- Add draft text display to construct page
- Add document.svg icon

jurb committed:
keep out search engines

jurb committed:
Ammend ignore

jurb committed:
Ammend ignore

jurb committed:
Add clear stores and navigation functionality to Bezwaar versturen button

- Add handleBezwaarVersturen function to clear API responses and session data
- Import clearApiResponses and goto for store clearing and navigation
- Connect onclick handler to ButtonSketchySmall component
- Convert ButtonSketchySmall to TypeScript with proper Props interface
- Fix TypeScript errors for component prop types

jurb committed:
Improve routing logic for high-confidence AI matches

- Add direct navigation to construct page when similarity > 0.5
- Find highest matching intent using reduce instead of max similarity
- Route to /1/construct/{intentcode} for confident matches
- Convert script to TypeScript for better type safety
- Maintain existing /1/choose route for low-confidence matches

jurb committed:
add figma screenshot

jurb committed:
Add TypeScript to remaining Svelte pages

- Add lang='ts' to /2/+page.svelte
- Add lang='ts' to /2/record/+page.svelte with proper type annotations
- Add lang='ts' to /test-endpoints/+page.svelte
- Improves type safety and consistency across all pages

jurb committed:
Clean up layout file

- Remove commented-out code blocks
- Remove empty style block
- Simplify layout to essential components only

jurb committed:
Implement centralized error handling system

- Add errorStore.ts with centralized error management
- Create ErrorDisplay.svelte component for UI error display
- Add ErrorDisplay to layout for app-wide error handling
- Update recording and construct pages to use centralized error handling
- Replace console.error with user-friendly error messages
- Add TypeScript to layout file for consistency

jurb committed:
Add loading states to construct page

- Add isSendingChat state for API call loading
- Display loading indicator with animated dots during chat API calls
- Improve user experience with visual feedback during async operations
- Add CSS animations for loading dots with staggered timing

jurb committed:
Make similarity threshold configurable

- Create configStore.ts for centralized app configuration
- Replace hardcoded 0.5 threshold with configurable value
- Add helper functions for threshold management
- Include documentation for developers on how to modify threshold
- Improves maintainability and allows runtime configuration

jurb committed:
Improve semantic structure and design of home page

- Add proper semantic HTML with main, nav, section, footer elements
- Replace basic links with descriptive concept cards
- Add MainMessage component for consistent hero section
- Include detailed descriptions for each concept
- Add responsive grid layout for concept navigation
- Implement hover effects and accessibility improvements
- Add TypeScript support and proper component imports
- Include development disclaimer footer

jurb committed:
Update homepage content to Dutch

- Change all English text to Dutch to match application language
- Update hero section, concept descriptions, and footer text
- Maintain consistent Dutch language throughout the application
- Keep semantic structure and styling unchanged

jurb committed:
Add error display test buttons for debugging

- Add test buttons for error, warning, info, and API error types
- Allow easy testing of error display system in development
- Position buttons with existing debug components
- Include Dutch error messages for consistency

jurb committed:
Fix layout: make left panel scrollable with fixed recording section

- Remove fixed height constraint from .app to allow page scrolling
- Make chat-container scrollable with overflow-y: auto
- Move recording section outside chat container to fix at bottom
- Add border and styling to separate recording from chat area
- Allow whole page to scroll to reach API debugger components
- Keep recording button always visible at bottom of left panel

jurb committed:
Move record button 50px higher from bottom

- Add margin-bottom: 50px to recording-section
- Provides better visual spacing from bottom edge

jurb committed:
Adjust record button position - move additional 20px higher

- Increase margin-bottom from 50px to 70px
- Fine-tune positioning for better UX

jurb committed:
Add 15px top margin to pill container

- Add margin-top: 15px to .pill-container
- Provides spacing above the 'Parkeerboete bezwaarformulier' pill

jurb committed:
Add comprehensive localization implementation plan

- Create detailed todo list for multilingual app conversion
- Include phase-by-phase implementation strategy
- List all components and files requiring translation
- Define Whisper integration points for auto-language detection
- Outline technical decisions and success metrics

jurb committed:
Implement Phase 1 of localization system

- Install and configure svelte-i18n library
- Create translation files for Dutch (nl), English (en), and French (fr)
- Set up i18n configuration and language store for manual switching
- Initialize i18n system in app layout with onMount
- Convert homepage to use translation keys with $_() function
- Add LanguageSwitcher component for testing language changes
- Implement language persistence with localStorage
- All Phase 1 foundation tasks completed successfully

jurb committed:
Configure localization to always default to Dutch

- Remove browser language detection, always start with Dutch (nl)
- Add setLanguageFromAPI() function for future Whisper integration
- Add confidence threshold logic (commented for future use)
- Add shouldSwitchLanguage() helper for comparing detected vs current
- Manual language switches still persist, API switches are temporary
- Ready for programmatic language switching from API responses

jurb committed:
Fix SSR initialization error for svelte-i18n

- Add async initialization with waitLocale() to ensure translations load
- Add hooks.server.ts for server-side i18n initialization
- Add loading state check to prevent rendering before translations ready
- Handle both client and server-side rendering properly
- Resolves 'Cannot format a message without first setting the initial locale' error

jurb committed:
Fix header visibility and remove duplicate language switcher

- Add homepage route (/) to header language dropdown visibility
- Remove separate LanguageSwitcher component from homepage
- Simplify layout loading logic to prevent header disappearing
- Header now shows logo and language dropdown on homepage
- Use existing header dropdown instead of separate component

jurb committed:
Fix header to always show on all pages including homepage

- Add debug logging to see actual route IDs
- Simplify header logic: always show logo and language dropdown on non-construct pages
- Remove specific route conditions that were preventing header from showing
- Header should now be visible on homepage and all other routes

jurb committed:
Add robust i18n initialization with proper loading states

- Add proper loading screen with logo while i18n initializes
- Prevent rendering components before translations are ready
- Add error handling and initialization guards
- Use both i18nReady state and isLoading check
- Wait specifically for Dutch locale to load
- Show branded loading screen instead of header flash

jurb committed:
Remove debug console logs

- Remove route debugging log from Header component
- Remove i18n initialization success log
- Keep error logging for production debugging

jurb committed:
Connect header language dropdown to localization system

- Import language store and setLanguage function
- Make dropdown reactive to current language state
- Handle language switching for supported languages (nl, en, fr)
- Add feedback for unsupported languages with alert
- Auto option defaults to Dutch as expected
- Remove unused selectedOption state management

jurb committed:
Fix TypeScript and Svelte 5 compliance in Header

- Add lang='ts' to script tag for proper TypeScript support
- Add proper event typing for handleDropdownChange function
- Update deprecated on:change to onchange event attribute
- Add HTMLSelectElement type assertion for event target

jurb committed:
Add complete localization to /1 route pages

- Add concept1 translation keys to all language files (nl, en, fr)
- Update /1/+page.svelte with MainMessage and CalloutBubble translations
- Update /1/choose/+page.svelte with title and subtitle translations
- Add parameter interpolation for transcript in choose page subtitle
- Update /1/construct/[slug]/+page.svelte with AI loading and sender translations
- Add back button translation in Header component
- All /1 route pages now fully support language switching

jurb committed:
Add localization support to components

- Updated ButtonSketchy, ButtonSketchySecondary, and ButtonSketchySmall to use translations for default text
- Added TypeScript interfaces and proper typing for all button components
- Updated construct page form elements to use translations
- Added new translation keys for buttons and form elements across all languages
- Components now fallback to translated defaults while still accepting explicit text props

jurb committed:
Add comprehensive error message localization

- Added translation keys for all error messages, warnings, and info messages
- Updated ErrorDisplay component to use translated aria-label for dismiss button
- Enhanced errorStore with translation-aware helper functions:
  - showTranslatedError/Warning/Info for translated messages
  - Updated handleApiError to use translation keys for context
- Updated all error usage throughout the app:
  - Record page now uses translated error messages
  - Construct page test buttons use translated messages
  - API error contexts now use translation keys
- All three languages (nl, en, fr) updated with comprehensive error translations

jurb committed:
Adjust button font size

jurb committed:
remove bind from checklist, gave error and does not need to be reactive anyway

jurb committed:
Add internationalization to concept 2 routes

Replace hardcoded Dutch text with translation keys in /2 and /2/record pages to support multi-language functionality. Follows same i18n pattern as concept 1 routes.

jurb committed:
Translate tags

jurb committed:
Svelte5 syntax

kapsterdam committed:
All yap (scenario 2) endpoints

jurb commented:
First of all, fantastic etc etc!!! 🥳

Going through the README:

* There is still an IP in there which we should probably remove from our repo and cycle
* The command 

   ```
   curl -X POST http://128.251.225.11/yap/start \
  -H "Content-Type: application/json" \
  -d '{
    "transcript": "We willen op 14 september een buurt‑bbq ...",
    "intentcode": "block_party_subsidy"
   }'
   ```
   
   does not work, it expects `text` instead of `transcript` in the body.
   
 I will test the rest as soon as possible, great that we have the yap endpoint, fixed draft for /chat, unified request type /chat etc!

jurb committed:
Adjust api calls to changes backend

jurb committed:
connect yap to record page (accumulated transcript works! :))

jurb committed:
ignore my TODO for now

jurb committed:
Send /yap responses to store (and log store for debugging)

jurb committed:
Implement keyboard record on 'E' and push to talk

jurb committed:
Add hidden /test-devices page for microphone testing

- Create device enumeration page listing microphones, pointer devices, and keyboards
- Add real-time audio level visualization for each microphone
- Implement test buttons to identify which microphone is which
- Include audio stream management with Web Audio API
- Style with responsive layout and proper text wrapping for long device IDs

kapsterdam committed:
Logging env variables

kapsterdam committed:
Logging of env variables modified

kapsterdam commented:
Ja hier ook zie ik nu:

curl  http://128.251.225.11/pin

jurb committed:
Go to correct page

jurb committed:
WIP concept 2 pages

jurb committed:
spelling

jurb committed:
Transcribing animation for record button

kapsterdam committed:
/Analyse returns language flag and /chat returns user_text (transcription)

jurb committed:
Refactor recording functionality into reusable TranscriptionRecordingSection component

- Extract transcription recording logic from /2/record page into new TranscriptionRecordingSection component
- Add onStateChange and existingText props to SingleRecordingSection for better state management
- Improve component reusability and separation of concerns
- Add proper textarea styling for transcription input

jurb committed:
pnpm lint all the things

jurb committed:
add confetti

jurb committed:
Add end page to /1 with confetti

jurb committed:
correct flow for /1

jurb committed:
copy end and feedback flow to /2

jurb committed:
Built in 10 second timeout for transcribe after recording, inapp error message

jurb committed:
Update recording timeout to 20 seconds with internationalized error messages

- Changed timeout from 10 to 20 seconds in SingleRecordingSection component
- Updated error messages in English, Dutch, and French translations
- Updated fallback error message text
- Provides better user experience with longer timeout threshold

jurb committed:
Add 2-minute inactivity timer for concept routes

Implements automatic session reset after 2 minutes of user inactivity on all /1 and /2 concept screens. Timer clears API store and redirects users back to their concept root page.

jurb committed:
increase timeout to 30 seconds temporarily (message still says 20 sec)

jurb committed:
remove debug components

jurb committed:
Add sophisticated multilingual animated subtitle for concept 1

Implements language-aware subtitle animation that cycles between non-selected languages with smooth transitions. Features elastic bounce animations and reactive filtering based on current language selection.

jurb committed:
formatting

jurb committed:
lower threshold

jurb committed:
Implement bottom-anchored chat with auto-scroll and animations

Both chat interfaces now feel like real messaging apps with messages appearing at bottom, auto-scroll to latest content, and smooth slide-in animations for new messages.

jurb committed:
Update readme

jurb committed:
WIP: style improvements agents-chat

jurb committed:
Remove border radius from most places 🥲

kapsterdam committed:
Cloud whisper + updated prompts + user_text is returned in chat

jurb committed:
slugify with a proper library

jurb committed:
WIP agents-chat styling

jurb committed:
Normal font in call-out bubble, size tweaks

jurb committed:
Get all matches

jurb committed:
Dynamic text in pill

jurb committed:
Change terug in 'Opnieuw beginnen' and wipe apistore

jurb committed:
Right intent above the draft

jurb committed:
repetitive things that should be in a layout file

jurb committed:
much shorter delay

jurb committed:
copy

jurb committed:
slightly longer confetti

jurb committed:
add 'jouw agent'

kapsterdam committed:
Updated BERT model and /chat system message

jurb committed:
Change language via API response 🎉

jurb committed:
Remove confindence stuff, does not exist

jurb committed:
Add complete feedback system with Svelte 5 compatibility

- Implement FeedbackForm component with voice recording and text input
- Add FinalThankYou component with auto-redirect using $effect()
- Create feedback API endpoint with session tracking
- Add internationalization for Dutch, English, and French
- Convert from onMount to $effect rune for Svelte 5 compliance
- Update feedback pages to use new reusable components
- Add final-thanks routes for both concepts
- Include visual assets (checkmark icon, Figma screenshot)

jurb committed:
feedback formatting

jurb committed:
Implement complete voice feedback transcription system

- Add /api/feedback-transcribe endpoint for audio transcription
- Transform FeedbackForm to use transcription-only workflow like /2/record:
  * Read-only textarea displays transcribed text from recordings
  * SingleRecordingSection with 'e' key support for audio recording
  * Incremental transcription with + symbol for multiple recordings
  * Submit button appears only when transcription text exists
- Update SingleRecordingSection to support feedback-transcribe endpoint:
  * Send existing text for incremental transcription
  * Handle language detection from transcribe responses
- Simplify /api/feedback to only accept JSON text submissions
- Fix feedback endpoint error handling to properly forward backend status codes
- Clear API responses when leaving final-thanks page to ensure clean state
- Update apiHelpers to support new feedback-transcribe endpoint type

jurb committed:
Simplify agent-chat

jurb committed:
fix close button to right concept

jurb committed:
Tighten up agent-chat

jurb committed:
even better agent chat

jurb committed:
Privacy statements

jurb committed:
adjust callout text

jurb committed:
Comment out unavailable languages

jurb committed:
More translations

jurb committed:
Reset language to dutch on concept resets

jurb committed:
Remove unused styles

jurb committed:
Add sparkles to 'toekomst!'

jurb committed:
Tweak sparkles

jurb committed:
Even more confetti tweaks

jurb committed:
Fix record button click events in feedback form

Enable pointer events for record button by adding CSS rule to override
the disabled pointer events on the button wrapper. This fixes the issue
where clicks were not registering on the record button in /2/feedback.

jurb committed:
Fix record button clickability by removing negative z-index

Remove z-index: -2 from .app class in FeedbackForm that was putting
the entire feedback form behind other elements, preventing clicks
on the record button. Also reverted incorrect CSS fix from previous commit.

jurb committed:
Fix confetti visibility by adjusting z-index

Change confetti container z-index from -1 to 1 so it appears above
the main content now that the app no longer has negative z-index.

jurb committed:
Position confetti behind heading using DOM order

Move confetti container before the heading in DOM structure so it
naturally appears behind the text without using z-index. Remove
z-index property to avoid layering complexity.

jurb committed:
Fix processSteps to react to language changes

Convert processSteps from static array to derived rune so the
internationalized text updates when the language changes.

jurb committed:
text in one block

jurb committed:
Fix avatar chat message

jurb committed:
Internationalize hardcoded strings in construct page

Add translation keys for "formulier"/"form" and "Betreft:"/"Subject:"
to all language files (nl, en, fr) and replace hardcoded Dutch strings
in construct page with i18n calls. Now these strings will switch
correctly when users change language.

jurb committed:
Internationalize progress text in ChecklistCard

Add translation keys for "compleet"/"complete"/"terminé" to all
language files and replace hardcoded Dutch string in ChecklistCard
component. Progress percentage text now switches correctly with
language changes.

jurb committed:
Internationalize ChecklistCard subtitle default value

Add translation keys for the default subtitle text in all language
files and update ChecklistCard component to use the internationalized
version. The subtitle now switches correctly between languages:
- Dutch: "Vermeld de volgende gegevens voor een complete aanvraag:"
- English: "Provide the following details for a complete request:"
- French: "Fournissez les détails suivants pour une demande complète:"

jurb committed:
Make ChecklistCard subtitle reactive to language changes

Replace static prop default with derived rune that reactively updates
when language changes. Uses subtitle prop if provided, otherwise falls
back to translated text that updates when language switches.

jurb committed:
More i18n

jurb committed:
Replace loading indicator with ChatMessage in construct page

Replace custom loading indicator with ChatMessage component showing
"schrijft antwoord..." (writing response...) using the existing
concept2.writingResponse i18n key. This provides consistent UI with
the agents-chat page and proper internationalization support.

Remove unused CSS for the old loading indicator.

jurb committed:
Add optional language parameter to all API endpoints

- Update apiHelpers.ts to accept language parameter and pass to all endpoints
- Modify all 7 API server routes to forward language to backend
- Update components to pass current language from languageStore
- Preserve existing language detection functionality from API responses
- Enable backend to receive user's language preference for localized responses

jurb committed:
Merge branch 'feature/app-frontend' into app-scaffold-translation-stub

jurb committed:
Implement comprehensive i18n support for AI assistant backend

Add full internationalization support for Dutch (nl), English (en), and French (fr) languages across all backend components.

Key changes:
- Create translation infrastructure with JSON files for all three languages
- Add i18n utility module with language detection and normalization
- Update all API endpoints (/analyze, /chat, /yap routes) to handle language parameters
- Internationalize AI system prompts (BURGER_SYSTEM, GEMEENTE_SYSTEM)
- Localize intent definitions and response messages
- Update Pydantic models with translated field descriptions
- Add comprehensive test suite for i18n functionality

Technical implementation:
- Language parameters accepted via query string, JSON body, or FormData
- Automatic fallback to Dutch for invalid/missing language codes
- Maintains backward compatibility with existing API contracts
- AI prompts dynamically generated based on user language
- Intent embeddings remain language-agnostic for consistent matching

The backend now fully supports the language parameters already being sent by the internationalized frontend, enabling a seamless multilingual user experience.

jurb committed:
we also need to create yap_log if it doesn't exist?

jurb committed:
More translations

jurb committed:
More translations

jurb committed:
translate intents according to recognized language

jurb committed:
please not here already

jurb committed:
Translate prompts for more reliable localized responses, fix language flag not going out to /chat

jurb committed:
Correct name on the chat bubble

jurb committed:
Correct name on the chat bubble

kapsterdam committed:
cloud whisper now working correctly

kapsterdam committed:
tiny mistake with exception handling

kapsterdam committed:
Merge pull request #4 from Amsterdam-AI-Team/app-scaffold-translation-stub

App scaffold translation stub

kapsterdam committed:
added feedback endpoint

kapsterdam committed:
Merge pull request #5 from Amsterdam-AI-Team/app-scaffold-translation-stub

added feedback endpoint

kapsterdam committed:
fixed feedback endpoint with feedback field instead of text

kapsterdam committed:
Merge pull request #6 from Amsterdam-AI-Team/app-scaffold-translation-stub

fixed feedback endpoint with feedback field instead of text

kapsterdam committed:
added support for json payload

kapsterdam committed:
Merge pull request #7 from Amsterdam-AI-Team/app-scaffold-translation-stub

added support for json payload

jurb committed:
Merge branch 'feature/app-frontend' into feature/app-scaffold

Commit:
Fixed transcription with Azure

kapsterdam committed:
Changed prompt

jurb committed:
Disabled ability to type in the textarea

jurb committed:
turned header logo into hard reset

jurb committed:
Update agent success / concluding message

jurb committed:
Increase inactivity timer by ridiculous amount

kapsterdam committed:
Fixed draft in chat

kapsterdam committed:
Merge branch 'feature/app-scaffold' into feature/app-scaffold-bugfix-draft

kapsterdam committed:
Merge pull request #8 from Amsterdam-AI-Team/feature/app-scaffold-bugfix-draft

Fixed draft in chat

kapsterdam committed:
fixed arguments for make_chain

kapsterdam committed:
Merge pull request #9 from Amsterdam-AI-Team/feature/app-scaffold-bugfix-draft

fixed arguments for make_chain

kapsterdam committed:
fixed make_chain

kapsterdam committed:
Merge pull request #10 from Amsterdam-AI-Team/feature/app-scaffold-bugfix-draft

fixed make_chain

jurb committed:
New draft data structure for concept 1

jurb committed:
fix grammar mistake

jurb committed:
lower threshold for intent matching

jurb committed:
Fix YAP generation and transcription issues

Restore internationalization functionality that was lost during recent merges:
- Add missing model creation imports to templates.py
- Update _yap_generate() to use language-specific models and prompts
- Fix _yap_check_finished() to use localized completion keywords
- Restore local Whisper fallback when TRANSCRIPTION_API_KEY is missing

Fixes TypeError: _yap_generate() takes 3 positional arguments but 4 were given
Fixes NameError: name 'GemeenteTurnModel' is not defined
Fixes RuntimeError: TRANSCRIPTION_API_KEY not set

jurb committed:
dont parse and message anymore

jurb committed:
translation

jurb committed:
Revert "Fix YAP generation and transcription issues"

This reverts commit d535b3b1c877a21e164589f7deba7c08f79cfb74.
